pkg/build/reaper/reaper.go

```diff
@@ -30,78 +31,93 @@ type BuildConfigReaper struct {
 
 // Stop deletes the build configuration and all of the associated builds.
 func (reaper *BuildConfigReaper) Stop(namespace, name string, timeout time.Duration, gracePeriod *kapi.DeleteOptions) error {
-	noBcFound := false
-
-	// Add deletion pending annotation to the build config
-	err := unversioned.RetryOnConflict(unversioned.DefaultRetry, func() error {
-		bc, err := reaper.oc.BuildConfigs(namespace).Get(name)
-		if kerrors.IsNotFound(err) {
```

Comment:
i think there was a point in time at which we would still attempt to delete builds that were associated with the buildconfig, even if we couldn't find the buildconfig.

that behavior was recently changed such that we just error out if we can't find the buildconfig and do nothing further.  So yeah, this logic is probably no longer needed.